packages/api/cms-api/src/user-management/auth/permission-check.ts

```diff
@@ -0,0 +1,30 @@
+import { EntityClass, EntityManager } from "@mikro-orm/core";
+import { CustomDecorator, SetMetadata } from "@nestjs/common";
+import { Request } from "express";
+
+import { ContentScope } from "../../common/decorators/content-scope.interface";
+import { CurrentUser } from "../current-user";
+import { UserPermission } from "../entities/user-permission.entity";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type PermissionCheckArgs<ArgsType = any> = {
+    args: ArgsType;
+    getScopeFromEntity: (entityClass: EntityClass<object>, id: string) => Promise<ContentScope>;
+    user: CurrentUser;
+    entityManager: EntityManager;
+    request: Request;
+};
+
+export type PermissionCheckOptions = {
+    entityClass?: EntityClass<object>;
+    isAllowed?: (args: PermissionCheckArgs) => Promise<boolean> | boolean;
+    getScopeFromMeta?: (args: PermissionCheckArgs) => Promise<ContentScope>;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getScopeFromArgs?: (args: any) => ContentScope;
```

Comment:
getScopeFromArgs should probably be not part of @PermissionCheck so it can be used by other part of the library:

- ScopeGuard (if we decide to keep it)
- Scope dependent build triggers
- Scope in user action log (do we need this?)

Reply:
You could use the ContentScopeService for it: https://github.com/vivid-planet/comet/blob/next/packages/api/cms-api/src/content-scope/content-scope.service.ts#L28